### DISCUSSION TITLE
🖐️ Caching docker layers using dagger

### DISCUSSION THREAD

**Discussion by mlathara** [dagger/dagger#12462] [Help]
### What happened? What did you expect to happen?

We're using dagger Go SDK for our CI pipelines in GitHub, and I am having trouble understanding how to enable caching docker layers. We're specifically using [Container.Build](https://pkg.go.dev/dagger.io/dagger#Container.Build) to build the image and while it does take a [ContainerBuildOpts](https://pkg.go.dev/dagger.io/dagger#ContainerBuildOpts) argument, that seems to just allow build args. 

I was hoping to be able to use [GHA](https://docs.docker.com/build/cache/backends/gha/), maybe by passing `cache-from` and `cache-to` to `Container.Build`. Sorry if I'm misunderstanding how this is supposed to work -- I looked through the docs, cookbook and issue and didn't see a reference to anything like this.

**Comment by mlathara** [comment]
I hadn't checked the discord -- but there is some discussion about this there: https://discord.com/channels/707636530424053791/1122941618056614048/1131263862981804042

My summary is that what I'm asking for isn't supported (anymore), and that there are scaling issues with using GHA anyway. There's also a closed source solution for caching that isn't currently available for GitHub. Please correct me if I'm missing something. If it's about right, then feel free to close this issue, thanks!

**Comment by d3rp3tt3** [comment]
Hi @mlathara, we'll have an update on distributed caching later this week in the Dagger community call. There are a few threads in Discord on caching, and the most recent one is here: https://discord.com/channels/707636530424053791/1151898625945915454. Hope to see you there. :)
